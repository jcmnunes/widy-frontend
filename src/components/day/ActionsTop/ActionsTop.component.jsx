import React from 'react';
import PropTypes from 'prop-types';
import Logout from '../../auth/Logout/Logout.container';
import { Dropdown, StyledDropdown, Button, RoundButton } from '../../UI';
import {
  IconCheveronDown,
  IconCog,
  IconSurvey,
  IconUser,
  IconUserCircle,
  StyledActionsTop,
} from './ActionsTop.styles';

const ActionsTopComponent = ({ noDays }) => (
  <StyledActionsTop>
    <RoundButton iconAfter={<IconSurvey />} disabled={noDays}>
      Widy Report
    </RoundButton>
    <Dropdown
      trigger={<RoundButton iconBefore={<IconCheveronDown />} iconAfter={<IconUserCircle />} />}
      position="right"
    >
      <StyledDropdown>
        <Button intent="dropdown" iconBefore={<IconUser />} block>
          Edit profile
        </Button>
        <Button intent="dropdown" iconBefore={<IconCog />} block>
          Settings
        </Button>
        <Logout />
      </StyledDropdown>
    </Dropdown>
  </StyledActionsTop>
);

ActionsTopComponent.propTypes = {
  noDays: PropTypes.bool.isRequired,
};

export default ActionsTopComponent;
