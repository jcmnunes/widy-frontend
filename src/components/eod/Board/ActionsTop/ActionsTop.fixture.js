export default {
  logout: {
    loading: false,
    error: '',
  },
};
