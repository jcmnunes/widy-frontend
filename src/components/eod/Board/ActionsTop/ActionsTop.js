import React, { Component } from 'react';
import styled from 'styled-components';
import { IconPresentation, IconCheveronDown, IconUserCircle } from '../../../../icons/Icons';
import { RoundButton } from '../../../UI';

const StyledActionsTop = styled.div`
  display: flex;
  flex-direction: row;
  align-items: center;

  button {
    margin-left: 12px;
  }
`;

class ActionsTop extends Component {
  render() {
    return (
      <StyledActionsTop>
        <RoundButton iconAfter={<IconPresentation />}>Standup</RoundButton>
        <RoundButton iconBefore={<IconCheveronDown />} iconAfter={<IconUserCircle />} />
      </StyledActionsTop>
    );
  }
}

export default ActionsTop;
