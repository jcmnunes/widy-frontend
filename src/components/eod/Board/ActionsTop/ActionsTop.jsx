import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components/macro';
import Logout from '../../../auth/Logout/Logout.container';
import {
  IconCheveronDown,
  IconCog,
  IconPresentation,
  IconSurvey,
  IconUser,
  IconUserCircle,
} from '../../../../icons/Icons';
import { Dropdown, StyledDropdown, Button, RoundButton } from '../../../UI';

const StyledActionsTop = styled.div`
  display: flex;
  flex-direction: row;
  align-items: center;

  & > * {
    margin-left: 12px;
  }
`;

const ActionsTop = ({ noDays, theme }) => (
  <StyledActionsTop>
    <RoundButton
      iconAfter={<IconSurvey primaryColor={theme.neutral200} secondaryColor={theme.neutral600} />}
      disabled={noDays}
    >
      Widy Report
    </RoundButton>
    <RoundButton iconAfter={<IconPresentation />} disabled={noDays}>
      Standup
    </RoundButton>
    <Dropdown
      trigger={<RoundButton iconBefore={<IconCheveronDown />} iconAfter={<IconUserCircle />} />}
      position="right"
    >
      <StyledDropdown>
        <Button intent="dropdown" iconBefore={<IconUser />} block>
          Edit profile
        </Button>
        <Button intent="dropdown" iconBefore={<IconCog />} block>
          Settings
        </Button>
        <Logout />
      </StyledDropdown>
    </Dropdown>
  </StyledActionsTop>
);

ActionsTop.propTypes = {
  noDays: PropTypes.bool.isRequired,
  theme: PropTypes.shape({
    [PropTypes.string]: PropTypes.string,
  }).isRequired,
};

export default ActionsTop;
