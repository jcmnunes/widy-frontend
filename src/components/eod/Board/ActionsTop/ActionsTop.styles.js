import styled from 'styled-components/macro';
import Icon from '../../../../icons/Icon';

export const StyledActionsTop = styled.div`
  display: flex;
  flex-direction: row;
  align-items: center;

  & > * {
    margin-left: 12px;
  }
`;

export const IconCheveronDown = styled(Icon).attrs({
  icon: Icon.CHEVERON_DOWN,
})`
  .primary {
    fill: ${props => props.theme.neutral600};
  }
`;

export const IconSurvey = styled(Icon).attrs({
  icon: Icon.SURVEY,
})`
  .primary {
    fill: ${props => props.theme.neutral200};
  }

  .secondary {
    fill: ${props => props.theme.neutral600};
  }
`;

export const IconPresentation = styled(Icon).attrs({
  icon: Icon.PRESENTATION,
})`
  .primary {
    fill: ${props => props.theme.neutral600};
  }

  .secondary {
    fill: ${props => props.theme.neutral200};
  }
`;

export const IconUserCircle = styled(Icon).attrs({
  icon: Icon.USER_CIRCLE,
})`
  .primary {
    fill: ${props => props.theme.neutral200};
  }

  .secondary {
    fill: ${props => props.theme.neutral600};
  }
`;

export const IconUser = styled(Icon).attrs({
  icon: Icon.USER,
})`
  .primary {
    fill: ${props => props.theme.neutral200};
  }

  .secondary {
    fill: ${props => props.theme.neutral600};
  }
`;

export const IconCog = styled(Icon).attrs({
  icon: Icon.COG,
})`
  .primary {
    fill: ${props => props.theme.neutral200};
  }

  .secondary {
    fill: ${props => props.theme.neutral600};
  }
`;
