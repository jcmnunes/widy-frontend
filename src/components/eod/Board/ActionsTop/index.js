export { default } from './ActionsTop.container';
