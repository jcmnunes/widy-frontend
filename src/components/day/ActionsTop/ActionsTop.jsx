import React from 'react';
import PropTypes from 'prop-types';
import { Dropdown, DropdownItem, IconButton } from '@binarycapsule/ui-capsules';
import { StyledActionsTop } from './ActionsTop.styles';

const ActionsTop = ({ isLogoutLoading, logoutRequest }) => (
  <StyledActionsTop>
    <IconButton text="Widy report" icon="SURVEY" isRound hasBackground onClick={() => {}} />
    <Dropdown
      placement="right"
      trigger={<IconButton hasBackground hasChev icon="USER_CIRCLE" isRound />}
    >
      <DropdownItem
        text="Log out"
        isLoading={isLogoutLoading}
        icon="LOGOUT"
        handleAction={logoutRequest}
        closeOnAction={false}
      />
    </Dropdown>
  </StyledActionsTop>
);

ActionsTop.propTypes = {
  isLogoutLoading: PropTypes.bool.isRequired,
  logoutRequest: PropTypes.func.isRequired,
};

export default ActionsTop;
