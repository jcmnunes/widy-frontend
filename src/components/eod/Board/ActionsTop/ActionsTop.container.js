import { connect } from 'react-redux';
import ActionsTop from './ActionsTop';
import logoutThunk from '../../../../thunks/logout';

const mapStateToProps = state => ({
  logout: state.auth.logout,
});

export default connect(
  mapStateToProps,
  { logoutThunk },
)(ActionsTop);
