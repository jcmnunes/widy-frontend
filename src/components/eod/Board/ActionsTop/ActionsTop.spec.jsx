import React from 'react';
import { shallow } from 'enzyme';

import ActionsTop from './ActionsTop';
import fixture from './ActionsTop.fixture';

describe('EOD Components', () => {
  describe('ActionsTop', () => {
    const props = {
      ...fixture,
      logout: jest.fn(),
    };

    it('should render as expected, without crashing', () => {
      const wrapper = shallow(<ActionsTop {...props} />);
      expect(wrapper).toMatchSnapshot();
    });
  });
});
