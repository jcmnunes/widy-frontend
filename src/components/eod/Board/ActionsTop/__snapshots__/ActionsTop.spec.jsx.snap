// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EOD Components ActionsTop should render as expected, without crashing 1`] = `
<styled.div>
  <RoundButton
    disabled={false}
    iconAfter={
      <IconPresentation
        primaryColor="#52616D"
        secondaryColor="#CBD2D8"
        size={24}
      />
    }
    iconBefore={null}
    loading={false}
  >
    Standup
  </RoundButton>
  <Dropdown
    closeOnAction={false}
    position="right"
    trigger={
      <RoundButton
        disabled={false}
        iconAfter={
          <IconUserCircle
            primaryColor="#52616D"
            secondaryColor="#CBD2D8"
            size={24}
          />
        }
        iconBefore={
          <IconCheveronDown
            primaryColor="#52616D"
            size={24}
          />
        }
        loading={false}
      />
    }
  >
    <styled.div>
      <Button
        block={true}
        disabled={false}
        iconAfter={null}
        iconBefore={
          <IconUser
            primaryColor="#CBD2D8"
            secondaryColor="#52616D"
            size={24}
          />
        }
        intent="dropdown"
        loading={false}
        onClick={null}
        size="medium"
        type="button"
      >
        Edit profile
      </Button>
      <Button
        block={true}
        disabled={false}
        iconAfter={null}
        iconBefore={
          <IconDoorExit
            primaryColor="#CBD2D8"
            secondaryColor="#52616D"
            size={24}
          />
        }
        intent="dropdown"
        loading={false}
        onClick={[MockFunction]}
        size="medium"
        type="button"
      >
        Log out
      </Button>
    </styled.div>
  </Dropdown>
</styled.div>
`;
